DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/Cell.h

```diff
@@ -22,62 +22,160 @@ namespace emcal
+
+  /// \brief Get cell amplitude
+  /// \return cell Amplitude
+  ///
+  /// Set getEnergy for more information
   Double_t getAmplitude() const { return getEnergy(); }
 
-  void setType(ChannelType_t ctype);
-  ChannelType_t getType() const;
+  /// \brief Set the type of the cell
+  /// \param ctype Type of the cell (HIGH_GAIN, LOW_GAIN, LEDMON, TRU)
+  void setType(ChannelType_t ctype) { mCellWord.mCellStatus = static_cast<uint16_t>(ctype); }
+
+  /// \brief Get the type of the cell
+  /// \return Type of the cell (HIGH_GAIN, LOW_GAIN, LEDMON, TRU)
+  ChannelType_t getType() const { return static_cast<ChannelType_t>(mCellWord.mCellStatus); }
+
+  /// \brief Check whether the cell is of a given type
+  /// \param ctype Type of the cell (HIGH_GAIN, LOW_GAIN, LEDMON, TRU)
+  /// \return True if the type of the cell matches the requested type, false otherwise
+  bool isChannelType(ChannelType_t ctype) const { return getType() == ctype; }
+
+  /// \brief Mark cell as low gain cell
+  void setLowGain() { setType(ChannelType_t::LOW_GAIN); }
+
+  /// \brief Check whether the cell is a low gain cell
+  /// \return True if the cell type is low gain, false otherwise
+  Bool_t getLowGain() const { return isChannelType(ChannelType_t::LOW_GAIN); }
+
+  /// \brief Mark cell as high gain cell
+  void setHighGain() { setType(ChannelType_t::HIGH_GAIN); }
 
-  void setLowGain();
-  Bool_t getLowGain() const;
+  /// \brief Check whether the cell is a high gain cell
+  /// \return True if the cell type is high gain, false otherwise
+  Bool_t getHighGain() const { return isChannelType(ChannelType_t::HIGH_GAIN); };
 
-  void setHighGain();
-  Bool_t getHighGain() const;
+  /// \brief Mark cell as LED monitor cell
+  void setLEDMon() { setType(ChannelType_t::LEDMON); }
 
-  void setLEDMon();
-  Bool_t getLEDMon() const;
+  /// \brief Check whether the cell is a LED monitor cell
+  /// \return True if the cell type is LED monitor, false otherwise
+  Bool_t getLEDMon() const { return isChannelType(ChannelType_t::LEDMON); }
 
-  void setTRU();
-  Bool_t getTRU() const;
+  /// \brief Mark cell as TRU cell
+  void setTRU() { setType(ChannelType_t::TRU); }
 
-  void setLong(ULong_t l);
-  ULong_t getLong() const { return mBits.to_ulong(); }
+  /// \brief Check whether the cell is a TRU cell
+  /// \return True if the cell type is TRU, false otherwise
+  Bool_t getTRU() const { return isChannelType(ChannelType_t::TRU); }
 
   void PrintStream(std::ostream& stream) const;
 
  private:
-  std::bitset<40> mBits;
+  union {
+    struct {
+      uint16_t mTowerID : 15;   ///< bits 0-14   Tower ID
+      int16_t mTime : 11;       ///< bits 15-25: Time (signed, can become negative after calibration)
+      uint16_t mEnergy : 14;    ///< bits 26-39: Energy
+      uint16_t mCellStatus : 2; ///< bits 40-41: Cell status
+      uint16_t mZerod : 6;      ///< bits 42-47: Zerod
+    };
+    char mDataWors[6]; ///< Bit representation
+  } mCellWord;         ///< Data word
```

Comment:
1st of all, root cannot store anonymous structs and you actually don't need such a union. 
Then, your struct has 2B alignment (largest data member) and the bit count of your data members are crossing 2B boundaries. So the compiler will add padding and as a result the sizeof(Cell) will be 8B instead of expected 6. With the bitset sized you have you need to use packig. Then, you need to init the vaulues.
So, you should use smt. like:

````
class Cell
{
 public:
  Cell() {
    auto tmp = reinterpret_cast<uint16_t*>(&mCellWord);
    tmp[0] = tmp[1] = tmp[2] = 0;
  }

  struct  __attribute__((packed)) CellWord {
      uint16_t mTowerID : 15;   ///< bits 0-14   Tower ID
      int16_t mTime : 11;          ///< bits 15-25: Time (signed, can become negative after calibration)
      uint16_t mEnergy : 14;    ///< bits 26-39: Energy
      uint16_t mCellStatus : 2; ///< bits 40-41: Cell1 status
      uint16_t mZerod : 6;      ///< bits 42-47: Zerod
    } mCellWord;

   ClassDefNV(Cell, 1);
};
````

Reply:
I see, OK, I will change this.